### PR TITLE
fix: DHIS2-7712: DRAFT Review

### PR DESCRIPTION
**Pull request by maikelarabori**
Unit tests are not added yet. I just want another point of view regarding this code fix.

Thanks

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/security/DefaultAnalyticsSecurityManager.java [hunk 1]
```diff
@@ -80,6 +85,38 @@
...
+            {
+                final List<CategoryOption> categoryOptions = category.getCategoryOptions();
+
+                for ( Iterator<CategoryOption> it = categoryOptions.iterator(); it.hasNext(); )
+                {
+                    final CategoryOption categoryOption = it.next();
+
+                    if ( !hasDataReadPermissionFor( categoryOption ) )
+                    {
+                        it.remove();
```
Comment (luciano-fiandesio) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/security/DefaultAnalyticsSecurityManager.java]: I think you can simplify this nested loop with:
```
for ( Category category : programCategories )
{
   category.getCategoryOptions().removeIf(categoryOption -> !hasDataReadPermissionFor(categoryOption));
}
```

**Comment by maikelarabori** [comment]
Hi @luciano-fiandesio, I'm marking this branch as ready for review. Please let me know if you have any other comments for this PR.

Thanks,
Maikel

**Comment by maikelarabori** [comment]
Hi @luciano-fiandesio , do you think I can merge this PR. It seems like I have permission now.

Thanks,
Maikel

**Comment by maikelarabori** [comment]
Absolutely, let me check why it's happening.

**Pull request merged**